components/DeleteButton.tsx

```diff
@@ -13,12 +12,16 @@ export default function DeleteButton({ itemId, title }: DeleteButtonProps) {
     throw new Error('item is undefined');
   }
 
-  const router = useRouter();
+  const handleDelete = async () => {
+    try {
+      await deleteItems(itemId);
+      window.location.reload();
+    } catch (error) {
+      console.error('Failed to delete item:', error);
+    }
+  };
   return (
-    <button
-      className='button button-rounded '
-      onClick={() => deleteItems(itemId).then(router.refresh)}
-    >
+    <button className='button button-rounded ' onClick={handleDelete}>
```

Comment:
Keeping arrow functions out of props is a generally a good practice to keep - I like this change 👌